Docs: Restructure testing section

kylegach opened the pull request:
Closes #31173 

## What I did

- Storybook Test-first
- test-runner only documented in dedicated page
- Update usage of "component test"
- Remove Interactions essentials page

## Checklist for Contributors

### Testing

#### Manual testing

1. Open this dedicated deploy preview: https://deploy-preview-281--storybook-frontpage.netlify.app/docs/9/writing-tests
1. Sync this branch with locally running docs
1. Compare with existing docs: https://storybook.js.org/docs/9/writing-tests

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

<!-- greptile_comment -->

## Greptile Summary

This PR comprehensively restructures Storybook's testing documentation to emphasize Storybook Test over test-runner, with clearer organization and updated terminology.

- Renamed and consolidated testing documentation under `writing-tests/`, with test-runner and Vitest addon moved to `integrations/`
- Updated all documentation links from `component-testing.mdx` to `interaction-testing.mdx` across framework templates
- Added new comprehensive guides for interaction testing, snapshot testing, and accessibility testing
- Removed redundant content including `essentials/interactions.mdx` and component testing snippets
- Reorganized a11y configuration snippets with clearer examples and better TypeScript support



<!-- /greptile_comment -->

vanessayuenn commented on docs/writing-tests/index.mdx:
```diff
@@ -6,23 +6,290 @@ sidebar:
+
+In our point of view, the goal is not to get 100% coverage and fill every gap, but rather to have a barometer to helps you judge code/test changes, and for the gaps to inform you of key states or interactions that are untested. For example, if you’re building a prototype you might skip testing altogether, whereas in a critical app you might want to dot every i and cross every t.
+
+Running coverage analysis can slow down your test runs, so it is turned off by default. To activate coverage, press the edit button (pencil icon) in the testing module and check the coverage checkbox.
```
We don't have the pencil icon anymore!

vanessayuenn commented on docs/writing-tests/index.mdx:
```diff
@@ -6,23 +6,290 @@ sidebar:
+
+Accessibility is not only the right thing to do, but it is increasingly mandated. For example, the [European accessibility act](https://ec.europa.eu/social/main.jsp?catId=1202) is scheduled to go into law in June 2025. Similarly in the US, laws like the [Americans with Disabilities Act (ADA)](https://www.ada.gov/) and [Section 508 of the Rehabilitation Act](https://www.section508.gov/) apply to m ...
+
+To activate accessibility checks alongside your component tests, press the edit button (pencil icon) in the testing module, then check the Accessibility checkbox.
```
Pencil icon no more 🙅🏼 ✏️

kylegach committed:
Restructure testing section

- Storybook Test-first
- test-runner only documented in dedicated page
- Update usage of "component test"
- Remove Interactions essential page

kylegach committed:
Address feedback

kylegach committed:
Fix remaining snippets

- `your-renderer` -> `your-framework`
- `@storybook/test` -> `storybook/test`
- `canvasElement` -> `canvas`
- Destructure `userEvent` from play fn context, instead of import
- Remove `ts`; Rename `ts-4-9` -> `ts`

kylegach committed:
Update coverage guidance

- Prioritize calculating all test coverage, not just stories

kylegach committed:
Update static assets

jonniebigodes commented on docs/writing-tests/integrations/vitest-addon.mdx:
```diff
@@ -62,7 +60,7 @@ For some project setups, the `add` command may be unable to automate the addon a
 1. Make sure Vite and Vitest are configured in your project.
 1. Configure Vitest to use [browser mode](https://vitest.dev/guide/browser/).
-1. Install the addon, `@storybook/addon-vitest`, in your project and [register it in your Storybook configuration](http://storybook.js.org/docs/addons/install-addons#manual-installation).
+1. Install the addon, `@storybook/addon-vitest`, in your project and [register it in your Storybook configuration](../addons/install-addons.mdx#manual-installation).
```
Another broken link here (this is deeply nested, should be another one level up). When you can re-check this page again for these issues.

jonniebigodes commented on docs/writing-tests/integrations/vitest-addon.mdx:
````diff
@@ -185,56 +183,15 @@ This screenshot shows how you can run your Vitest tests in VSCode using the [Vit
-```
+However, to have the test output link to your published Storybook on test failures, you need to provide the [`storybookUrl` option](#storybookurl) in the plugin configuration. Please reference the [detailed example in the Testing in CI guide](./in-ci.mdx#21-debugging-test-failures-in-ci).
 
 ## How it works
````
When you're able, can you take another pass at this page specifically from this section (i.e., How it works) and onto  the one below (i.e., Configuring tests), and restructure it as we're:
- Repeating some information including the testing types and how to do it in the Storybook UI.
- The user needs to read through the middle of the page to learn about the test types he/she can perform and how to toggle them.

jonniebigodes reviewed (changes requested):
Thanks for taking the time to put this together, this is shaping up nicely. I've left some items for you to look into when you can including missing snippets, some broken links, and some structural changes that need to happen before this gets merged.

Let me know once you've taken a pass at this so that I can take another look.

kylegach committed:
Merge branch 'next' into docs-update-testing-section

* next: (36 commits)
  Remove `withActions` and `parameters.action.handles`
  dont collect telemetry on selectedStoryCount
  Docs: Misc SB 9 updates
  flush any pending test case results when test run completes
  Update CHANGELOG.md for v8.6.14 [skip ci]
  Increase test resilience by adding retry logic to web-components properties tests
  Apply requested changes
  reduce CI failures due to vue3 testrunner
  Bump version from "9.0.0-rc.1" to "9.0.0-rc.2" [skip ci]
  CSF-Tools: Add support for existing node imports and improve import handling
  Write changelog for 9.0.0-rc.2 [skip ci]
  Bump version from "9.0.0-rc.0" to "9.0.0-rc.1" [skip ci]
  Update CHANGELOG.md for v8.6.13 [skip ci]
  Extract to constant
  Fix undefined error
  Test: Patch HTMLElement.prototype.focus method for settable focus in tests
  Force a mininum height/width for the targetable area for each highlight
  ReactNativeWeb: JSX should be set to automatic
  Remove duplicate notification dot on sidebar buttons on mobile
  Update docs
  ...

kylegach commented on docs/writing-tests/in-ci.mdx:
```diff
@@ -0,0 +1,467 @@
+  test:
+    runs-on: ubuntu-latest
+    container:
+      # Make sure to grab the latest version of the Playwright image
+      # https://playwright.dev/docs/docker#pull-the-image
+      image: mcr.microsoft.com/playwright:v1.50.1-noble
```
I updated the image version, but disagree that the current link, https://playwright.dev/docs/docker#pull-the-image, should be replaced with that one. The only reason we link out at all is to encourage people to use the latest image. We're provided our own CI guidance, and linking to Playwright's is overwhelming, imo.

kylegach commented on docs/writing-tests/accessibility-testing.mdx:
```diff
@@ -27,258 +25,198 @@ If you want to check accessibility for your stories using the [addon](https://st
+<Video src="../_assets/writing-tests/sb-test-a11y-run.mp4" />
 
-### Disable a11y tests with the test runner
+If you're using the [test-runner](./integrations/test-runner.mdx), you can run your accessibility tests in the terminal or in CI environments.
```
Wouldn't a callout draw _more_ attention? How does that align with downplaying it?

Rather than a callout, I instead added some headings for "With the Vitest addon" and "With the test-runner", to clarify this section further.

kylegach commented on docs/writing-tests/index.mdx:
```diff
@@ -6,23 +6,242 @@ sidebar:
-* [Test runner](./test-runner.mdx) to automate test execution
+_The full installation instructions, including project requirements, are available in the [Vitest addon documentation](./integrations/vitest-addon.mdx#install-and-set-up)._
+
+Once your project is set up, you will see a testing widget in the bottom of your sidebar. After running tests, you will also see test status indicators on sidebar items. Additionally, many tests can be debugged with an addon panel.
```
I think either are grammatically correct, but "in" will be more expected by a majority of readers.

kylegach commented on docs/writing-tests/integrations/test-runner.mdx:
```diff
@@ -1,10 +1,20 @@
+<If renderer={['react', 'vue', 'svelte']}>
+<Callout variant="warning">
+
+The test runner has been superceded by the [Vitest addon](./vitest-addon.mdx), which offers the same functionality, powered by the faster and more modern [Vitest](https://vitest.dev/) browser mode. It also enables the full Storybook Test experience, allowing you to run interaction, accessibility, and visual tests from your Storybook app.
```
TIL!

kylegach commented on docs/_snippets/storybook-interactions-step-function.md:
I removed them because I didn't think they did anything to help readers understand the use of `step`.

You're totally right that I could further simplify these by only having `common` snippets. thanks.

kylegach committed:
Address feedback

- Fix broken links from:
    - /writing-tests/test-addon -> /writing-tests/integrations/vitest-addon
    - /writing-tests/vitest-addon -> /writing-tests/integrations/vitest-addon
    - /writing-tests/test-runner -> /writing-tests/integrations/test-runner
    - /writing-tests/import-stories-in-tests/stories-in-unit-tests -> /writing-tests/integrations/stories-in-unit-tests
    - /writing-tests/import-stories-in-tests/stories-in-end-to-end-tests -> /writing-tests/integrations/stories-in-end-to-end-tests
- Create/update necessary image and video assets
- Ensure new snippets have all necessary variations
    - Angular TS
    - Svelte & Svelte CSF, TS & JS
    - Web components, TS & JS
- Replace remaining usages of "your-renderer" with "your-framework"
- Replace remaining usages of `@storybook/<renderer>` with `@storybook/<framework>`
- Normalize comment text for "your-framework"
- Various organizational and prose improvements

Co-authored-by: jonniebigodes <joaocontadesenvolvimento@gmail.com>

kylegach committed:
Merge branch 'next' into docs-update-testing-section

* next: (43 commits)
  Assert updated state in play function
  Build: Provide cursor environment
  Bump version from "9.0.0-rc.3" to "9.0.0-rc.4" [skip ci]
  Write changelog for 9.0.0-rc.4 [skip ci]
  Update package.json files
  Addons: Add shim Storybook addons for previously removed addons
  Bump version from "9.0.0-rc.2" to "9.0.0-rc.3" [skip ci]
  inline Storybook Icon
  fix
  Write changelog for 9.0.0-rc.3 [skip ci]
  fix lint
  Update code/core/src/csf-tools/CsfFile.ts
  don't rely on redundant import.meta variables in animation utils
  don't externalise @storybook/icons in manager-api
  Small refactorings
  Fix annotation parsing for renamed exports
  add renamed export test
  Ensure vitest exclusions are relative to the project root, not cwd
  make save-from-controls e2e tests work with an object-type Control instead of a text-type
  fix unit tests
  ...

jonniebigodes commented on docs/writing-tests/integrations/test-runner.mdx:
When you're able, can you adjust this page and:
  - Move the test types (A11y, Coverage, Snapshot) upwards above the CI section
      - The rationale is that we're trying to make it clear that these are the test types that can be run with the test runner and that the CI section is more about setting up the test runner in CI, which is a different topic and can factor into the test types that are run locally.
  - Combine the Troubleshooting sections into a single section
    - I'm aware that the it will be a bit longer, but I think it will be easier to read, and follows our existing pattern.

jonniebigodes reviewed (changes requested):
Appreciate you addressing the feedback. Just took another pass and left some items to address before we can get this in.
Let me know once you've checked them and I can take another pass

kylegach committed:
Address feedback

- Re-add mistakenly removed snippets
- Restructure test-runner docs

kylegach committed:
Merge branch 'next' into docs-update-testing-section

* next:
  always clean coverage before (re)running
  Controls: Remove empty state video link
  Docs: Add Code panel page

kylegach committed:
Merge branch 'next' into docs-update-testing-section
